src/Compilers/CSharp/Portable/Binder/BinderFactory.cs

```diff
@@ -160,7 +160,7 @@ internal InMethodBinder GetRecordConstructorInMethodBinder(SynthesizedRecordCons
 
         internal Binder GetInRecordBodyBinder(RecordDeclarationSyntax typeDecl)
         {
-            Debug.Assert(typeDecl.Kind() is SyntaxKind.RecordDeclaration);
+            Debug.Assert(typeDecl.Kind() is SyntaxKind.RecordDeclaration or SyntaxKind.RecordStructDeclaration);
```

Comment:
I could remove the assertion altogether. But I found myself keeping it. Let me know if you want to remove.

Reply:
> I could remove the assertion altogether. But I found myself keeping it. Let me know if you want to remove.

Given the parameter type, it looks like the assert cannot fail. If that is the case, I would say there is no reason to have it.